### PR TITLE
Fix resume offline train logic. Add loading optimizer state

### PR DESCRIPTION
**Pull request by hanq-moreh**
## Motivation

Add loading optimizer state when resume training. 
Loading the optimizer state when resuming training is required to restore momentum, learning rate schedules, and other optimizer-specific statistics. Without it, the optimizer behaves as if starting from scratch.

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://sgl-fru7574.slack.com/archives/C09784E3EN6 to discuss your PR.

### CODE REVIEW

**Comment by zyksir** [comment]
This should already be added in this PR #152 , Still thank you for you commit!

**Comment by hanq-moreh** [comment]
> This should already be added in this PR #152 , Still thank you for you commit!

But, still in `train_eagle3_offline.py`, I can't find load_state_dict() of optimizer in main branch. Can you check it please?
